lib/api/AuthTokensController.cpp

```diff
@@ -24,6 +24,7 @@ namespace MAT_NS_BEGIN {
 
     status_t  AuthTokensController::SetTicketToken(TicketType type, char const* tokenValue)
     {
+        LOCKGUARD(m_lock);
```

Comment:
Do you expect SetTicketToken to be called after SDK bootstrapping, from multiple thread ? If so, we should also lock in the getter methods.

Reply:
Yes, `SetTicketToken` will be called in multiple threads after we get the logmanager instance. Do you mean we should lock in the `GetAuthTokensController` method?

`getLogManager` is already guarded with a lock:

```
static ILogManager* getLogManager(jlong nativeLogManager)
{
    std::lock_guard<std::mutex> lock(jniManagersMutex);
    if (nativeLogManager < 0 || nativeLogManager >= jniManagers.size())
    {
        return nullptr;
    }

    return jniManagers[nativeLogManager]->manager;
}
```

Reply:
This means we need to lock GetUserTokent(), GetDevicetokens(), GetTickets() methods too. And these methods are called in event upload hot-path. These locks will add the latency in the event upload.

Reply:
@maxgolov Do you see any issue in adding the mutex lock in GetUserTokent(), GetDevicetokens(), and GetTickets() methods - which are called in the hot path of event upload?

Reply:
You may need to apply a bit more holistic approach to mutex here:
- mutex works just for the moment while the reference is obtained. But then the collection is returned by reference -> allows for potential concurrent modification and breakage outside of the function call.
- `status_t AuthTokensController::Clear()` then also needs a mutex.

Ideally the methods should be copying the contents into collection passed by the caller. Then you can use a mutex to guarantee that the copy of this collection is done under the mutex. Otherwise the way as it's currently implemented in the PR - is all unreliable, unfortunately.

Reply:
+1